### PR TITLE
read more feature

### PR DESCRIPTION
**Pull request by MaYaNkKashyap681**
1). Added Box Shadow Around Post Box.
2). Added Read More Feature in the post -> If the caption is long, the read more button will be displayed and if the user clicks on the read more button, the whole caption will be displayed.
3). Added Comment Icon.
4). On hovering the user profile, the circle turns black and the image scales up a little.

Screenshot (113)
Screenshot (114)

Screenshot (115)
Screenshot (116)

### CODE REVIEW

File: src/components/Post.jsx [hunk 1]
```diff
@@ -1,11 +1,11 @@
 import { useEffect, useState } from "react";
 import FavoriteBorderIcon from "@mui/icons-material/FavoriteBorder";
 import FavoriteOutlinedIcon from "@mui/icons-material/FavoriteOutlined";
+import CommentIcon from "@mui/icons-material/Comment";
 import { red } from "@mui/material/colors";
 import SentimentSatisfiedAltOutlinedIcon from "@mui/icons-material/SentimentSatisfiedAltOutlined";
 import MoreHorizOutlinedIcon from "@mui/icons-material/MoreHorizOutlined";
 import DeleteTwoToneIcon from "@mui/icons-material/DeleteTwoTone";
-import Scroll from "../reusableComponents/Scroll";
```
Comment (narayan954) [inline src/components/Post.jsx:8]: why are you removing this import, it's needed for the scroll component!

File: src/components/Post.jsx [hunk 2]
```diff
@@ -19,9 +19,6 @@ import {
   DialogContent,
   DialogContentText,
   useMediaQuery,
-  Box,
```
Comment (narayan954) [inline src/components/Post.jsx:22]: these 3 imports are not needed to be removed as well

File: src/components/Post.jsx [hunk 3]
```diff
@@ -315,130 +326,50 @@ function Post(prop) {
...
+              }}
+            >
+              View All comments
+            </Button>
             <DialogBox
               open={isCommentOpen}
               onClose={handleCommentClose}
               title="All Comments"
             >
-              <Box sx={{ flexGrow: 1 }}>
```
Comment (narayan954) [inline src/components/Post.jsx:333]: line 333 to 441 is also needed

**Review by narayan954 — CHANGES_REQUESTED** [review]
Hi @MaYaNkKashyap681 Some older changes have also been changed incorrectly in this new commit, can you please fix them? :) If you face any problem lmk

File: src/components/Post.jsx [hunk 4]
```diff
@@ -315,130 +338,148 @@ function Post(prop) {
...
 
-        {/* View all comments dialog Box */}
         {comments.length ? (
           <>
-            <Button onClick={setisCommentOpen}>View All comments</Button>
+          <Button
+              onClick={setisCommentOpen}
+              startIcon={<CommentIcon />}
+              sx={{
+                backgroundColor: "transparent",
```
Comment (narayan954) [inline src/components/Post.jsx:353]: backgroundColor property is duplicated, please look into it :)

File: src/components/Post.jsx [hunk 5]
```diff
@@ -315,130 +338,148 @@ function Post(prop) {
         <div className="post__text">
           {caption && postHasImages && (
             <>
-              <strong>{username} </strong>
-              {caption}
+              <strong style={{ color: "royalblue" }}>#{username} </strong>
```
Comment (narayan954) [inline src/components/Post.jsx:341]: Also Is # needed?

Comment (MaYaNkKashyap681) [reply src/components/Post.jsx:341]: i think it is ok if you want i can remove it.

**Review by narayan954 — APPROVED** [review]
lgtm!

File: src/components/Post.jsx (hunk 5 above)
Comment (narayan954) [reply src/components/Post.jsx]: fine

**Pull request merged**